### PR TITLE
refactor(model): clean up a few things, i.e., parameters for some methods

### PR DESCRIPTION
**Pull request by anarchuser**
1. Removes the "list of objects", reverting back to the old state with three lists for individuals, food and obstacles, respectively
2. Changes return type and parameters of Individual::act
  * Returns an `Action` instead of an `Action *` - actions are small enough to pass by value
  * Takes a smart pointer (`std::unique_ptr<T>`) of a list of objects
3. Adds Constructors taking Points as position for all object classes, to reduce parameters being passed and for nicer usage while actually constructing them with points

### CODE REVIEW

File: src/controller/controller.cpp [hunk 1]
```diff
@@ -40,9 +40,7 @@ bool Controller::simulate() {
 
         auto visibles = world.getObjectsAround(individual->getPosition(), individual->getVision().length);
         auto action = individual->act(visibles);
-        std::cout << action->toString() << std::endl;
-        delete action;
```
Comment (xeruf) [inline src/controller/controller.cpp:44]: why remove the delete?

**Review by xeruf — APPROVED** [review]

File: src/controller/controller.cpp (hunk 1 above)
Comment (anarchuser) [reply src/controller/controller.cpp:44]: I refactored the system to make the delete obsolete

Comment (anarchuser) [reply src/controller/controller.cpp:44]: by passing the action by value, not by reference, to be precise

**Comment by xeruf** [comment]
@anarchuser why did you not add the PR description into the commit message? Also, please do not use something like i.e. in the commit summary

**Pull request merged**